src/commands/kubeconfig.ts

```diff
@@ -0,0 +1,248 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { retryWithSleep } from "../common/retry";
+import { authenticate } from "../drivers/auth";
+import { guard } from "../drivers/firestore";
+import { print2 } from "../drivers/stdio";
+import { getAwsConfig } from "../plugins/aws/config";
+import {
+  awsCloudAuth,
+  profileName,
+  requestAccessToCluster,
+  getAndValidateK8sIntegration,
+} from "../plugins/kubeconfig";
+import { ensureEksInstall } from "../plugins/kubeconfig/install";
+import { exec, stricmp } from "../util";
+import { writeAwsConfigProfile, writeAwsTempCredentials } from "./aws/files";
+import yargs from "yargs";
+
+export type KubeconfigCommandArgs = {
+  cluster: string;
+  role: string;
+  resource?: string;
+  reason?: string;
+  requestedDuration?: string;
+};
+
+export const kubeconfigCommand = (yargs: yargs.Argv) =>
```

Comment:
So, the reason I went with this approach is because it seemed to me that `p0 request` was intended to be a fairly thin wrapper around the P0 API, sharing basically the same code path as Slack and MS Teams. Some of the things you mention, like mapping args for `p0 kubeconfig` to the actual request args, I think could actually be beneficial—for example, providing a more stable interface that we could keep consistent even if we have to update the request arguments. Plus, here I'm doing some extra validation before hitting the backend for things like the requested role; would we still want things like that folded into `p0 request`? But I also see your point around `p0 kubeconfig` not being obvious, though I feel like the way `p0 request` is used is not entirely obvious either.

But yeah, let's discuss this tomorrow for sure.

Reply:
Just to follow-up from the discussion earlier: We'll keep this as a separate `p0 kubeconfig` command for now and solicit customer feedback to determine if there's another name that makes more sense and/or if this should just be merged with the `p0 request k8s resource` command.